receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -234,6 +223,23 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, logger *slog.L
 	return nil
 }
 
+func (r *pReceiver) initScrapeOptions() *scrape.Options {
+	opts := &scrape.Options{
+		PassMetadataInContext: true,
+		ExtraMetrics:          r.cfg.ReportExtraScrapeMetrics,
+		HTTPClientOptions: []commonconfig.HTTPClientOption{
+			commonconfig.WithUserAgent(r.settings.BuildInfo.Command + "/" + r.settings.BuildInfo.Version),
+		},
+		EnableCreatedTimestampZeroIngestion: enableCreatedTimestampZeroIngestionGate.IsEnabled(),
+	}
+
+	if enableNativeHistogramsGate.IsEnabled() {
```

Comment:
nit: not sure why I wrote it in a separate `if`, probably to have smaller diff. I think it can be moved into the `opts` init as well above.